### PR TITLE
feat: trailingCommas option

### LINKED ISSUE #10: Add trailing comma configuration
Probably "always", "jsonc", and "never" options, with "never" being the default.

### PR DESCRIPTION
**Pull request by Zamiell**
Closes #10, #14.

Hello.

This is a WIP progress PR for a `trailingCommas` option.
However, I am stuck at this line:

```rs
let in_jsonc = false; // How to determine this?
```

In other words, from within the plugin (in the "generate.rs" file), how do I determine whether or not the current file being formatted has a ".json" extension or a ".jsonc" extension?

### CODE REVIEW

**Comment by Zamiell** [comment]
@dsherret Sorry for the ping, do you know if this project still being maintained?

**Comment by dsherret** [comment]
> @dsherret Sorry for the ping, do you know if this project is still being maintained?

Did you take the time to look at when the last commit was to main or where this project is used before asking that question? I have too many projects to maintain and miss a ton of GitHub notifications. I never look at or read draft prs.

> how do I determine whether or not the current file being formatted has a ".json" extension or a ".jsonc" extension?

You will need to add a file path parameter to format_text and use the file_path here: https://github.com/dprint/dprint-plugin-json/blob/ead41c5972c89f47dcacf63718a4c3dfaf8dacf2/src/wasm_plugin.rs#L53

Thanks!

**Comment by Zamiell** [comment]
> I never look at or read draft prs.

Ah, ok, that explains it, thanks. Is there a preferred way to contact you then if I have a question in the future when I am doing a draft PR?

> You will need to add a file path parameter to format_text and use the file_path here:

Thank you, that got me in the right direction. The PR is ready for review now.

File: README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # dprint-plugin-json
 
-[![](https://img.shields.io/crates/v/dprint-plugin-json.svg)](https://crates.io/crates/dprint-plugin-json) [![CI](https://github.com/dprint/dprint-plugin-json/workflows/CI/badge.svg)](https://github.com/dprint/dprint-plugin-json/actions?query=workflow%3ACI)
+[![crates.io version](https://img.shields.io/crates/v/dprint-plugin-json.svg)](https://crates.io/crates/dprint-plugin-json) [![CI](https://github.com/dprint/dprint-plugin-json/workflows/CI/badge.svg)](https://github.com/dprint/dprint-plugin-json/actions?query=workflow%3ACI)
```
Comment (Zamiell) [inline README.md:3]: This is unrelated to the PR, but I noticed it was bugged (no alt-text) when I opened the file in VSCode.

**Comment by Zamiell** [comment]
@dsherret Do you know if there is anyone on the team that can review this PR? It's been sitting for over a month now.

**Comment by dsherret** [comment]
There's nobody else on the team. I'll get to it soon.

**Review by dsherret — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**